staging/src/k8s.io/apiserver/pkg/util/wsstream/conn.go

```diff
@@ -265,6 +267,18 @@ func (conn *Conn) handle(ws *websocket.Conn) {
 			klog.V(6).Infof("Frame is targeted for a reader %d that is not valid, possible protocol error", channel)
 			continue
 		}
+		// Half-close the stream when receiving a frame with the STDIN channel id, but
```

Comment:
This API Server code is vendored into the container runtime: `containerd`, and it is the WebSocket server. This code handles the half-close signal, closing the STDIN stream.

Reply:
so, this means that for having the end to end websocket path we need al runtimes to support this, right?

Reply:
With the new `StreamTranslator`, we will not have to deal with any component upstream of the API Server. This still gives us the most important functionality (WebSockets between k8s clients and the API Server), while significantly reducing the scope of our effort.